### PR TITLE
Dev branch

### PR DESCRIPTION
**Pull request by amosmachora**
I have made a lot of changes majorly on the cors configuration and Safaricom IP addresses whitelisting.

### CODE REVIEW

**Review by Murithijoshua — APPROVED** [review]
LGTM

File: app/api/mpesa/stk-push/route.ts [hunk 1]
```diff
@@ -2,24 +2,25 @@ import { NextRequest, NextResponse } from "next/server";
...
-    amount: string;
-    phoneNumber: string;
-    transactionDesc: string;
-    callbackURL: string;
-  } = {
+  const requestBody: RequestBody = {
     accountReference,
     amount,
-    callbackURL: `${MPESA_APP_BASE_URL}/api/mpesa/stk-push-callback`,
+    callbackURL: `https://billing.kenyahmis.org/api/mpesa/stk-push-callback`,
```
Comment (ojwanganto) [inline app/api/mpesa/stk-push/route.ts:21]: Can we have this part of the environment variables?

Comment (amosmachora) [reply app/api/mpesa/stk-push/route.ts:21]: Its possible but since we know where our app is hosted i feel we can still explicitly define the URL.

Comment (ojwanganto) [reply app/api/mpesa/stk-push/route.ts:21]: We don't want to expose that on github :). It's just a way of keeping safe

Comment (amosmachora) [reply app/api/mpesa/stk-push/route.ts:21]: Okay. I understand now. I will change that ASAP

Comment (amosmachora) [reply app/api/mpesa/stk-push/route.ts:21]: should i do the same for the Kenyaemr 3.x repo too?

**Pull request merged**